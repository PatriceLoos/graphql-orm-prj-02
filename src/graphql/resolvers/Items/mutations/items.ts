import ItemsRepository from '../../../../entities/Items/ItemsRepository';
import { Resolver, Mutation, Arg, ID } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import CreateItemInput from '../types/inputs/CreateItem';
import Item from '../types/objects/Items'

@Resolver()
export default class CreateItemsResolver {
  @InjectRepository(ItemsRepository)
  private itemsRepository!: ItemsRepository;

  @Mutation(() => Item, { description: 'Create a new item in the database' })
  async createItem(
    @Arg('input', () => CreateItemInput)
    input: CreateItemInput
  ): Promise<Item> {
    console.log(">> Create item resolver",input)
    return await this.itemsRepository.createItem(input);
  }
  @Mutation(() => Item, { description: 'Update existing item in the database' })
  async updateitem(
    @Arg('id', () => ID ) id:string,
    @Arg('input', () => CreateItemInput)
    input: CreateItemInput,
  ): Promise<Item> {
    console.log(`>> Update item ${id} resolver with ${Object.keys(input)}`)
    return await this.itemsRepository.updateItem(id,input);
  }

  @Mutation(() => Item, { description: 'delete item database' })
  async deleteitem(
    @Arg('id', () => ID ) id:string,
  ): Promise<any> {
    console.log(`>> Delete item ${id}.`)
    return await this.itemsRepository.delete(id).then((ret: { raw: any; }) =>{
      ret.raw
    });
  }
}
