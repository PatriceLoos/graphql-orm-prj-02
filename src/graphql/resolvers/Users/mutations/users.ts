import { Resolver, Mutation, Arg, ID } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import User from '../types/objects/Users';
import UserRepository from '../../../../entities/Users/UserRepository';
import CreateUserInput from '../types/inputs/CreateUser';

@Resolver()
export default class CreateUserResolver {
  @InjectRepository(UserRepository)
  private UserRepository!: UserRepository;

  @Mutation(() => User, { description: 'Create a new User in the database' })
  async createUser(
    @Arg('input', () => CreateUserInput)
    input: CreateUserInput
  ): Promise<User> {
    console.log(">> Create User resolver",input)
    return await this.UserRepository.createUser(input);
  }
  @Mutation(() => User, { description: 'Update existing User in the database' })
  async updateUser(
    @Arg('id', () => ID ) id:string,
    @Arg('input', () => CreateUserInput)
    input: CreateUserInput,
  ): Promise<User> {
    console.log(`>> Update User ${id} resolver with ${Object.keys(input)}`)
    return await this.UserRepository.updateUser(id,input);
  }

  @Mutation(() => User, { description: 'delete user database' })
  async deleteUser(
    @Arg('id', () => ID ) id:string,
  ): Promise<any> {
    console.log(`>> Delete User ${id}.`)
    return await this.UserRepository.delete(id).then(ret =>{
      ret.raw
    });
  }
}
