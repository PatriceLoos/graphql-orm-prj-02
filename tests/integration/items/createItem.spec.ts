import { Connection } from 'typeorm';
import { createTestClient } from 'apollo-server-testing';
import { createItemMutation } from './__gql__/mutations';
import {
  createTestConnection,
  createTestServer,
  clearTestDb,
} from '../../utils';

let testConnection: Connection;

beforeAll(async () => {
  testConnection = await createTestConnection();
});

beforeEach(async () => {
  await clearTestDb(testConnection);
});

afterAll(async () => {
  await testConnection.close();
});

describe('Create Items', () => {
  it('should register a new item in the database', async () => {
    const itemInput = {
      name: "cravache",
      address: "2020 ouille st, St-Louis, MS, 020102",
      description: "whip",
      brand: "Fischer",
      make: "Aerospace",
      model: "brown suede",
      size: "unisize",
      genderUsage: "n/a",
      type: "",
      identifier: "8787878",
      ownerEmail: "d@zone.com",
      dod: "08/29/2010",
      dop: "12/12/2001",
      imageUrl: "www.url.gov",
    };

    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: createItemMutation,
      variables: itemInput,
    });
    expect(res).toMatchSnapshot('New item');
  });
});
