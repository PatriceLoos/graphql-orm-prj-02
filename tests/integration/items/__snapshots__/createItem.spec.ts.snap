// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Items should register a new item in the database: New item 1`] = `
Object {
  "data": Object {
    "createItem": Object {
      "address": "2020 ouille st, St-Louis, MS, 020102",
      "brand": "Fischer",
      "description": "whip",
      "dod": "08/29/2010",
      "dop": "12/12/2001",
      "genderUsage": "n/a",
      "id": 1,
      "identifier": "8787878",
      "imageUrl": "www.url.gov",
      "make": "Aerospace",
      "model": "brown suede",
      "name": "cravache",
      "ownerEmail": "d@zone.com",
      "size": "unisize",
      "type": "",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
